versions/1/data_stream/fields/fields.spec.yml

```diff
@@ -6,66 +6,82 @@ spec:
   type: array
   items:
     type: object
+    # TODO: We should disable additionalProperties. There are properties used in
+    # elastic/integrations that have no purpose or are typos.
     additionalProperties: true
     properties:
       name:
-        description: Name of field
-        type: string
-      title:
-        description: Title of field
+        description: >
+          Name of field. Names containing dots are automatically split into
+          sub-fields.
         type: string
+        # TODO: Should we allow wildcards if they are not implemented in Fleet?
+        pattern: '^[_@A-Za-z0-9]+(\.[_@A-Za-z0-9]+)*$'
       type:
         description: Datatype of field
         type: string
         enum:
         - alias
-        - histogram
-        - constant_keyword
-        - text
-        - match_only_text
-        - keyword
-        - long
-        - integer
-        - short
+        - array
+        - binary
+        - boolean
         - byte
-        - double
-        - float
-        - half_float
-        - scaled_float
+        - constant_keyword
         - date
         - date_nanos
-        - boolean
-        - binary
-        - integer_range
-        - float_range
-        - long_range
-        - double_range
         - date_range
-        - ip_range
-        - group
+        - double
+        - double_range
+        - flattened
+        - float
+        - float_range
         - geo_point
-        - object
+        - group
```

Comment:
Fleet EPM has a `group-nested` type. Does that need added to the list? Why would we have both `group-nested` and `nested`?

https://github.com/elastic/kibana/blob/0695df649705c91ed28dbc73037aea878ce419f7/x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts#L120

Reply:
it was added 2 years ago in this pr, but doesn't say a lot of context about why: https://github.com/elastic/kibana/pull/64829

Reply:
I didn't try to follow the logic, but my take is that is seems like an internal-only value. So none of the integrations should use `group-nested` and it does not need to be in the spec.